sympy/polys/matrices/normalforms.py

```diff
@@ -13,6 +13,13 @@
 #  we should implement. See e.g. the Kannan-Bachem algorithm:
 #  <https://www.researchgate.net/publication/220617516_Polynomial_Algorithms_for_Computing_the_Smith_and_Hermite_Normal_Forms_of_an_Integer_Matrix>
 
+def extended_gcd(a, b):
+    """Return (gcd(a, b), x, y) such that a*x + b*y = gcd(a, b)"""
+    if not b:
+        return (a, 1, 0)
+    else:
+        d, x, y = extended_gcd(b, a % b)
+        return (d, y, x - (a // b) * y)
```

Comment:
Extended GCD is not needed to be implemented by hand because there is `gcdex`

Reply:
This algorithm should be implemented using domain types and DomainMatrix:
https://docs.sympy.org/latest/modules/polys/domainsintro.html

The gcdex operation can be found from the domain:
```python
In [10]: ZZ.gcdex(ZZ(12), ZZ(8))
Out[10]: (mpz(1), mpz(-1), mpz(4))

In [11]: R = QQ[x]

In [12]: R.gcdex(R(x**2 - 1), R(x - 1))
Out[12]: (0, 1, x - 1)
```

Reply:
I request to remove this function anyway